features/nanostack/coap-service/source/coap_security_handler.c

```diff
@@ -41,7 +42,14 @@ struct coap_security_s {
     mbedtls_ssl_config          _conf;
     mbedtls_ssl_context         _ssl;
 
-    mbedtls_ctr_drbg_context    _ctr_drbg;
+#if defined(MBEDTLS_CTR_DRBG_C)
+    mbedtls_ctr_drbg_context    _drbg;
+#elif defined(MBEDTLS_HMAC_DRBG_C)
+    mbedtls_hmac_drbg_context   _drbg;
+#else
+#error "CTR or HMAC must be defined for coap_security_handler!"
+#endif
```

Comment:
Why not always use the new Mbed OS global RNG and avoid use of DRBG entirely? The code should be much simplified if we always use the global RNG.